### DISCUSSION TITLE
Okunan roman sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/BilsemPython#54] [Soru Cevap] [answered]
Ali, bir romanın her gün bir önceki gün okuduğu sayfadan 5 sayfa fazlasını okumaktadır. İlk gün 10 sayfa okuyarak başlayan Alinin’ 1.000 sayfalık bir romanı kaç günde bitireceğini bulan programı kodlayınız.

(Önce algoritmasını kendi cümlelerinizle yazınız daha sonra kodunuzu ekleyiniz, kodunuzu kod bloğuna almayı unutmayınız)

**Comment by CinarSatir** [comment]
```python
#değişkenleri belirle
toplamsayfa = 10
gunsayi = 1
okunansayfa = 10
#gün sayısını arttırmak için döngü aç
while(toplamsayfa<1000):
    kalansayfa = 1000-toplamsayfa
    okunansayfa = okunansayfa+5
#son gün okunan sayfa sayısı hatalı oluyor onu degistirdim
    if kalansayfa < okunansayfa:
        okunansayfa = kalansayfa
    toplamsayfa = toplamsayfa+okunansayfa
    gunsayi = gunsayi+1
#bulunan değerleri yazdır
print("Ali",gunsayi,"gunde kitabı bitirir")
```

**Comment by QLNH9491** [comment]
`
top = 10
gun = 1
oku = 10
while(top<1000):
    kalan = 1000-top
    oku = oku+5
    if kalan < oku:
        oku = kalan
    top = top+oku
    gun = gun+1
print("Bay Ali kitabınızın biteceği gün",gun,"'dur. İyi günler.")
`

**Comment by EfeToprak** [comment]
```
toplamsayfa = 10
gunsayi = 1
okunansayfa = 10
#gün sayısını arttırmak için döngü aç
while(toplamsayfa<1000):
    kalansayfa = 1000-toplamsayfa
    okunansayfa = okunansayfa+5
#son gün okunan sayfa sayısı hatalı oluyor onu degistirdim
    if kalansayfa < okunansayfa:
        okunansayfa = kalansayfa
    toplamsayfa = toplamsayfa+okunansayfa
    gunsayi = gunsayi+1
#bulunan değerleri yazdır
print("Ali",gunsayi,"gunde kitabı bitirir")
```

**Comment by EfeMertIrez** [comment]
sayfa=10
toplam=10
gün=0
while (toplam 1000 den küçük eşit oldukça)
        sayfa=sayfa+5
        toplam=sayfa+toplam
        gün=gün+1
```
s=10
t=10
g=0
while t<=1000:
    s=s+5
    t=t+s
    g=g+1
print(g)
```

**Comment by slos8888** [comment]
```
toplamsayfa = 10
gunsayi = 1
okunansayfa = 10
while(toplamsayfa<1000):
    kalansayfa = 1000-toplamsayfa
    okunansayfa = okunansayfa+5
    if kalansayfa < okunansayfa:
        okunansayfa = kalansayfa
    toplamsayfa = toplamsayfa+okunansayfa
    gunsayi = gunsayi+1
print("Ali",gunsayi,"gunde kitabı bitirir")
```

**Comment by Efemert2010** [comment]
`t= 10
g = 1
os = 10
while(t<1000):
    k = 1000-t
    os = os+5
    if k < os:
        os = k
    t = t+os
    g = g+1
print("Ali",g,"gunde kitabı bitirir")`

**Comment by akgembi12** [comment, marked as answer]
hergün kaç sayfa okuyacağını ,toplam okunan sayfayı ve kaç kün geçtiğini tutacak değişkenleri tanımladım. Sonsuz bir döngü açtım ve okunan sayfa sayısını artırdım. Sonra günlük okunan sayfa sayısını 5 artırdım. okunan sayfa sayısı 1000den fazla ise döngüyü kırdım ve gün sayısını yazdırdım. Cevap ||19||
kod:
```py
gun=10
okunan=0
x=0
while True:
    okunan+=gun
    x+=1
    gun+=5
    if okunan>=1000:
        print(x,"günde biter")    
        break

```

**Comment by tolga0102** [comment]
a=alinin günlük okuduğu sayfa , x=gün
Ali'nin okuduğu sayfa sayısını kitap bitene kadar hep 5er arttırıp kitaba ekledim, bunu yaparken her artışta günü bir arttırdım. 
Kitap 1000den büyük olunca da gün sayısını yazdırdım.
```python
a=10
kitap=0
x=0
while kitap<1000:
    kitap+=a
    a+=5
    x+=1
print("Ali romanı",x,"günde bitirmiştir.")
```

**Comment by OmerMete** [comment]
hocam burda ilk toplam okunan sayfa,gün,ve okunan sayfa değişkenlerini ayarlıyoruz. sonra günü bulmak için while açıyoruz içerde gün ü buluyoruz ve günü yazdırıyoruz
```
ts=10
gun=1
os=10

while (ts<1000):
    os=os+5
    ts=ts+od
    gun=gun+1

print("ali bu kitabı",gun,"kadar günde bitirir")
```

**Comment by OmerB2** [comment]
Kitap sayfa sayısı belirtilir,
okunan sayfa belirtilir,
toplam okunan sayfa belirtilir,
gün belirtilir,
toplam okunan sayfa 1000'e eşit olana kadar çalışacak while döngüsü açılır,
okunan sayfaya 5 eklenir,
toplam okunan sayfaya okunan sayfa eklenir,
gün 1 arttırılır,
döngü bitince gün sayısı yazdırılır.
```
k=1000
os=10
tos=0
g=1
while tos<1001:
    os=os+5
    tos=tos+os
    g=g+1
print(g)
```

**Comment by Sadik1Dev** [comment]
```
gün = 1

toplam = 10

okunan = 10

while toplam <= 1000:

    gün = gün+1

    okunan = okunan + 5

    toplam = toplam+okunan
    
print(gün,"günde bitirir")

#gün sayısı ve toplam sayfa sayısını değişken olarak atadık
#sonra toplam sayfa sayısı 1000den fazla olduğunda dögüden çıkacak bir while döbüsü yazdık
#döngünün içersine sayfa sayısı 5 attırak ve günü 1 artırakcak kodu yazdık
#döngü bitincede günü yazdırdık
```

**Comment by AdaNilPoyraz** [comment]
```python
#o gün okuduğu sayfa sayısı (oku)
oku = 10
#toplam okunan sayfa sayısı (topoku)
topoku = 10
#gün sayısı (gun)
gun = 1
#kitap 1000 sayflık olduğu için toplam sayfa sayısının 1000den küçük olması gerek
while topoku < 1000:
    #hergün okunan sayfa sayısı 5 artırıyoruz
    oku += 5
    #toplam okunan sayfa sayısına ekliyoruz
    topoku += oku
    #gün sayısını artırıyoruz
    gun += 1
#en son da gün sayısını yazdırıyoruz
print("Ali kitabı",gun,"günde bitirir :)")

```

**Comment by efeozceylan** [comment]
Okumadığı sayfa sayısı üzerinden her gün 5 fazla eksilterek okumadığı sayfa kalmayacak şekilde döngü
oluşturdum.
```
os=10
kitap=1000
z=0
while kitap>0:
    kitap=kitap-os
    os=os+5
    z=z+1
print("Bu kitabi",z,"gunde bitirir.")
```
